File: topk-rs/tests/test_query_filter_regexp.rs
use test_context::test_context;
use topk_rs::query::field;
use topk_rs::query::filter;

mod utils;
use utils::dataset;
use utils::ProjectTestContext;

#[test_context(ProjectTestContext)]
#[tokio::test]
async fn test_query_regexp_match(ctx: &mut ProjectTestContext) {
    let collection = dataset::books::setup(ctx).await;

    let result = ctx
        .client
        .collection(&collection.name)
        .query(
            filter(field("_id").regexp_match("^cat", Option::<&str>::None)).limit(10),
            None,
            None,
        )
        .await
        .expect("could not query");

    assert_doc_ids!(result, ["catcher"]);
}

#[test_context(ProjectTestContext)]
#[tokio::test]
async fn test_query_regexp_match_with_flags(ctx: &mut ProjectTestContext) {
    let collection = dataset::books::setup(ctx).await;

    let result = ctx
        .client
        .collection(&collection.name)
        .query(
            filter(field("title").regexp_match("\\salchem", Some("i"))).limit(10),
            None,
            None,
        )
        .await
        .expect("could not query");

    assert_doc_ids!(result, ["alchemist"]);
}

File: topk-py/tests/test_query_filter_regexp.py
import pytest
from topk_sdk import data, error
from topk_sdk.query import field, filter, select, literal, not_

from . import ProjectContext
from .utils import dataset, doc_ids


def test_query_regexp_match(ctx: ProjectContext):
    collection = dataset.books.setup(ctx)

    result = ctx.client.collection(collection.name).query(
        filter(field("_id").regexp_match('^cat')).limit(10)
    )

    assert doc_ids(result) == {"catcher"}


def test_query_regexp_match_with_flags(ctx: ProjectContext):
    collection = dataset.books.setup(ctx)

    result = ctx.client.collection(collection.name).query(
        filter(field("title").regexp_match("\\salchem", "i")).limit(10),
    )

    assert doc_ids(result) == {"alchemist"}